yasa/heart.py

```diff
@@ -0,0 +1,177 @@
+        messages. The logging levels are 'debug', 'info', 'warning', 'error',
+        and 'critical'. For most users the choice is between 'info'
+        (or ``verbose=True``) and warning (``verbose=False``). Set this to True if you are getting
+        invalid results and want to better understand what is happening.
+
+    Returns
+    -------
+    epochs : :py:class:`pandas.DataFrame`
+        Output dataframe with stage and epoch number as index. The columns are
+
+        * 'start' : The start of the epoch, in seconds from the beginning of the recording.
+        * 'duration' : The duration of the epoch, in seconds.
+        * 'hr_mean': The mean heart rate (HR) across the epoch, in beats per minute (bpm).
+        * 'hr_std': The standard deviation of the HR across the epoch, in bpm
+        * 'hrv_rmssd': Heart rate variability across the epoch (RMSSD), in milliseconds.
+    rpeaks : dict
+        A dictionary with the detected heartbeats (R-peaks) indices for each epoch of each stage.
+        Indices are expressed as samples from the beginning of the epoch. This can be used to
+        manually recalculate the RR intervals, apply a custom preprocessing on the RR intervals,
+        and/or calculate more advanced HRV metrics.
+
+    Notes
+    -----
+    This function returns three cardiac features for each epoch: the mean and standard deviation of
+    the heart rate, and the root mean square of successive differences between normal heartbeats
+    (RMSSD). The RMSSD reflects the beat-to-beat variance in HR and is the primary time-domain
+    measure used to estimate the vagally mediated changes reflected in heart rate variability.
+
+    Heartbeat detection is performed with the SleepECG library: https://github.com/cbrnr/sleepecg
+
+    References
+    ----------
+    * Shaffer, F., & Ginsberg, J. P. (2017). An overview of heart rate variability metrics and
+      norms. Frontiers in public health, 258.
+    """
+    set_log_level(verbose)
+    is_sleepecg_installed()
+    from sleepecg import detect_heartbeats
+
+    if isinstance(hypno, type(None)):
+        logger.warning("No hypnogram was passed. The entire recording will be used.")
+        data = np.asarray(data, dtype=np.float64)
+        hypno = np.zeros(max(data.shape), dtype=int)
+        include = 0
+
+    # Safety check
+    (data, sf, _, hypno, include, _, n_chan, n_samples, _
+     ) = _check_data_hypno(data, sf, None, hypno, include, check_amp=False)
+    assert n_chan == 1, "data must be a 1D ECG array."
+    data = np.squeeze(data)
+
+    # Find periods of equal duration
+    epochs = hypno_find_periods(hypno, sf, threshold=threshold, equal_length=equal_length)
+    assert epochs.shape[0] > 0, f"No epochs longer than {threshold} found in hypnogram."
+    epochs = epochs[epochs["values"].isin(include)].reset_index(drop=True)
+    # Sort by stage and add epoch number
+    epochs.sort_values(by=['values', 'start'], inplace=True)
```

Comment:
Also not a big deal, but I'm under the impression the use of `inplace=True` is a big nono and that it's even in the (long-term) [process towards deprecation](https://github.com/pandas-dev/pandas/issues/16529). Unless you have a specific reason I'd suggest swapping out for the copy/overwrite version.

Reply:
(see also lines 130 and 175)